src/test/java/homework_4/singleton/SingletonTest.java

```diff
@@ -0,0 +1,15 @@
+package homework_4.singleton;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SingletonTest {
```

Comment:
you have 3 singleton implementations, but only 1 is tested 😢 
remove extra implementations or test every of them.